### PR TITLE
WorkflowConfigBuilder

### PR DESCRIPTION
**Pull request by danielnorberg**

### CODE REVIEW

**Comment by danielnorberg** [comment]
@honnix PTAL

**Comment by honnix** [comment]
bravo!

File: styx-common/src/main/java/com/spotify/styx/model/deprecated/WorkflowConfiguration.java [hunk 1]
```diff
@@ -118,10 +120,15 @@ public static Secret create(
       return new AutoValue_WorkflowConfiguration_Secret(name, mountPath);
     }
 
+    public static com.spotify.styx.model.WorkflowConfiguration.Secret create(
+        Secret secret) {
+      return com.spotify.styx.model.WorkflowConfiguration.Secret.create(
+          secret.name(), secret.mountPath());
+    }
+
     public static Optional<com.spotify.styx.model.WorkflowConfiguration.Secret> create(
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/model/deprecated/WorkflowConfiguration.java]: this method is not used by anyone now. can be deleted.

File: styx-common/src/main/java/com/spotify/styx/serialization/Json.java [hunk 2]
```diff
@@ -56,6 +57,7 @@ private Json() {
       .setPropertyNamingStrategy(SNAKE_CASE)
       .enable(ACCEPT_SINGLE_VALUE_AS_ARRAY)
       .disable(WRITE_DATES_AS_TIMESTAMPS)
+      .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/serialization/Json.java]: This seems to have bigger impact than annotating on class level. OK?

File: styx-local-files-source/src/main/java/com/spotify/styx/Yaml.java [hunk 3]
```diff
@@ -35,7 +36,8 @@
       .setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE)
       .enable(ACCEPT_SINGLE_VALUE_AS_ARRAY)
       .registerModule(new Jdk8Module())
-      .registerModule(new JavaTimeModule());
+      .registerModule(new JavaTimeModule())
+      .registerModule(new AutoMatterModule());
```
Comment (honnix) [inline styx-local-files-source/src/main/java/com/spotify/styx/Yaml.java]: Just as a reminder, this shall be applied to internal ScheduleSource as well. Maybe open an internal PR already now.

File: styx-common/src/main/java/com/spotify/styx/serialization/Json.java (hunk 2 above)
Comment (danielnorberg) [inline styx-common/src/main/java/com/spotify/styx/serialization/Json.java]: Yeah, maybe this should be another PR.

File: styx-local-files-source/src/main/java/com/spotify/styx/Yaml.java (hunk 3 above)
Comment (danielnorberg) [inline styx-local-files-source/src/main/java/com/spotify/styx/Yaml.java]: 👍

File: styx-common/src/main/java/com/spotify/styx/model/deprecated/WorkflowConfiguration.java (hunk 1 above)
Comment (danielnorberg) [inline styx-common/src/main/java/com/spotify/styx/model/deprecated/WorkflowConfiguration.java]: now it is =)

File: styx-common/src/main/java/com/spotify/styx/model/deprecated/WorkflowConfiguration.java [hunk 4]
```diff
@@ -80,14 +80,16 @@ public static WorkflowConfiguration create(
...
   }
 
-  public static com.spotify.styx.model.WorkflowConfiguration create(WorkflowConfiguration workflowConfiguration) {
-    return com.spotify.styx.model.WorkflowConfiguration
-        .create(workflowConfiguration.id(), workflowConfiguration.partitioning(), Optional.empty(),
-                workflowConfiguration.dockerImage(), workflowConfiguration.dockerArgs(),
-                Optional.of(workflowConfiguration.dockerTerminationLogging()),
-                Secret.create(workflowConfiguration.secret()),
-                Optional.empty(),
-                workflowConfiguration.resources());
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/model/deprecated/WorkflowConfiguration.java]: Seems `resources` Is missing here.
